src/main/java/no/ssb/subsetsservice/SubsetsControllerV2.java

```diff
@@ -756,27 +764,18 @@ else if (isFromDate && version.has(Field.VALID_UNTIL) && version.get(Field.VALID
                 String level = codeJsonNode.get(Field.LEVEL).asText();
                 String codeURN = classificationId+"_"+code+"_"+name+"_"+level;
                 ArrayNode classificationVersionsArrayNode = codeJsonNode.get(Field.CLASSIFICATION_VERSIONS).deepCopy();
-                if (codeMap.containsKey(codeURN)) {
-                    classificationVersionsArrayNode.addAll(codeMap.get(codeURN)); //TODO: This merger might cause duplicates
-                }
-                codeMap.put(codeURN, classificationVersionsArrayNode);
+                Set<String> classificationVersionsSet = new HashSet<>();
+                classificationVersionsArrayNode.forEach(e -> classificationVersionsSet.add(e.asText()));
+                classificationVersionsMap.merge(codeURN, classificationVersionsSet, (s1, s2) -> {s1.addAll(s2); return s1;});
             }
         }
 
         ArrayNode codesInRangeArrayNode = new ObjectMapper().createArrayNode();
 
-        for (String s : codeMap.keySet()) {
-            ArrayNode classificationVersionList = codeMap.get(s);
-
-            // Remove duplicates
-            for (int i = 0; i < classificationVersionList.size(); i++) {
-                JsonNode classificationVersion = classificationVersionList.get(i);
-                for (int i1 = i+1; i1 < classificationVersionList.size(); i1++) {
-                    if (classificationVersionList.get(i1).asText().equals(classificationVersion.asText())) {
-                        classificationVersionList.remove(i1);
-                    }
-                }
-            }
+        for (String s : classificationVersionsMap.keySet()) {
```

Comment:
I would instead iterate over each entrySet

Reply:
Thanks, that seems like the better way! Done.